### PR TITLE
Troubleshooting guides cleanup

### PR DESCRIPTION
**Pull request by colunira**
**Description**

Changes proposed in this pull request:

- Remove mentions about old Kyma versions (0.9.0)
- Delete guides for errors that no longer occur
- Update outdated guides

**Related issue(s)**

### CODE REVIEW

File: docs/service-mesh/10-02-no-access.md [hunk 1]
```diff
@@ -11,6 +11,21 @@ To fix this problem, restart the Pods of the Gateway.
...
+
+    ```
+   kubectl edit deployment -n istio-system istio-ingressgateway
+    ```
+   
+   Find the `ingress-sds` container and delete the `-distroless` suffix.
+
 2. Check all ports used by the Istio Ingress Gateway:
   ```
   kubectl exec -t -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system | grep "istio-ingressgateway" | awk '{print $1}') -- netstat -lptnu
```
Comment (Demonsthere) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
  ```bash
  kubectl exec -ti -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system -o name) -c istio-proxy -- netstat -lptnu
```

File: docs/service-mesh/10-02-no-access.md [hunk 2]
```diff
@@ -11,6 +11,21 @@ To fix this problem, restart the Pods of the Gateway.
...
+     ```
+     kubectl delete pod -l app=istio-ingressgateway -n istio-system
+     ```
+
+If the above solution doesn't work you need to change image of Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside of them. 
+
+1. Edit the Istio Ingress Gateway Deployment:
+
+    ```
+   kubectl edit deployment -n istio-system istio-ingressgateway
```
Comment (Demonsthere) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
    ```bash
    kubectl scale --replicas 0 -n istio-system deploy/istio-ingressgateway
    kubectl edit deployment -n istio-system istio-ingressgateway
    kubectl scale --replicas 1 -n istio-system deploy/istio-ingressgateway
```

File: docs/service-mesh/10-02-no-access.md [hunk 3]
```diff
@@ -11,6 +11,21 @@ To fix this problem, restart the Pods of the Gateway.
...
+    ```
+   
+   Find the `ingress-sds` container and delete the `-distroless` suffix.
```
Comment (Demonsthere) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
   Find the `istio-proxy` container and delete the `-distroless` suffix.
```

File: docs/service-mesh/10-02-no-access.md [hunk 4]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
+    kubectl get virtualservice --all-namespaces
+    ```
+
+2. Restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
+
+     ```bash
+     kubectl delete pod -l app=istio-ingressgateway -n istio-system
+     ```
+
+If the above solution doesn't work you need to change image of Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside of them.
```
Comment (kazydek) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
If this solution doesn't work, you need to change the image of the Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside them. 
```

File: docs/service-mesh/10-02-no-access.md [hunk 5]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
+    ```
+
+2. Restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
+
+     ```bash
+     kubectl delete pod -l app=istio-ingressgateway -n istio-system
+     ```
+
+If the above solution doesn't work you need to change image of Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside of them. 
+
```
Comment (kazydek) [inline docs/service-mesh/10-02-no-access.md]: Follow these steps:

File: docs/service-mesh/10-02-no-access.md [hunk 6]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
+1. Edit the Istio Ingress Gateway Deployment:
+
+    ```bash
+    kubectl scale --replicas 0 -n istio-system deploy/istio-ingressgateway
+    kubectl edit deployment -n istio-system istio-ingressgateway
+    kubectl scale --replicas 1 -n istio-system deploy/istio-ingressgateway
+    ```
+   
+   Find the `istio-proxy` container and delete the `-distroless` suffix.
```
Comment (kazydek) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
   2. Find the `istio-proxy` container and delete the `-distroless` suffix.
```

File: docs/service-mesh/10-02-no-access.md [hunk 7]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
 
 2. Check all ports used by the Istio Ingress Gateway:
```
Comment (kazydek) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
3. Check all ports used by the Istio Ingress Gateway:
```

File: docs/service-mesh/10-02-no-access.md [hunk 8]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
-  ```
-  kubectl exec -t -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system | grep "istio-ingressgateway" | awk '{print $1}') -- netstat -lptnu
-  ```
+
+    ```bash
+    kubectl exec -ti -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system -o name) -c istio-proxy -- netstat -lptnu
+    ```
 
 3. If ports `80` and `443` are not used, restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
```
Comment (kazydek) [inline docs/service-mesh/10-02-no-access.md]:
```suggestion
4. If ports `80` and `443` are not used, restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
```

File: docs/service-mesh/10-03-connection-refused.md [hunk 9]
```diff
@@ -3,7 +3,9 @@ title: Connection refused errors after upgrade
...
 ---
 
-Starting with Kyma 1.0, mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. Alternatively, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
+Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. If you don't want to use a sidecar, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
 
 - To enable sidecar injection for Pods of existing Services, restart them after upgrading to Kyma 1.0 or higher.
 - To create a DestinationRule for a Service, follow the official [Istio documentation](https://istio.io/docs/reference/config/networking/destination-rule/).
+
+>**TIP:** You can use a [`istioctl authn tls-check`](https://istio.io/docs/reference/commands/istioctl/#istioctl-authn-tls-check) command to find out where communication fails.
```
Comment (kazydek) [inline docs/service-mesh/10-03-connection-refused.md]:
```suggestion
>**TIP:** You can use the [`istioctl authn tls-check`](https://istio.io/docs/reference/commands/istioctl/#istioctl-authn-tls-check) command to find out where communication fails.
```

File: docs/service-mesh/10-02-no-access.md [hunk 10]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
+    ```
+   
+2. Find the `istio-proxy` container and delete the `-distroless` suffix.
+
+3. Check all ports used by the Istio Ingress Gateway:
+
+    ```bash
+    kubectl exec -ti -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system -o name) -c istio-proxy -- netstat -lptnu
+    ```
+
```
Comment (piotrmsc) [inline docs/service-mesh/10-02-no-access.md]: Hmm, previous steps mention to restart pod, if it doesn't help we modify image to use non distroless image which force another restart of pod, if mentioned command will show that port 443 is not opened what will another restart bring? IMO nothing :) In such case we need to find the root cause, mostly it is due to corrupted or missing certificate secret. Ingress will not open port 443 untill all mentioned secrets with certificates (They are defined on Gateways) are properly loaded by `istio-sds` container.

So we check logs from `istio-sds`container in istio-ingressgateway pod, look for errors + ensure secrets exists in istio-system namespace. if they exist check/validate  certificate if it's not for example string `aaaaa`. `istio-sds` in most of the cases report any problems it find with secrets referenced on Gateways (CRs) :)

File: docs/service-mesh/10-03-connection-refused.md [hunk 11]
```diff
@@ -3,7 +3,9 @@ title: Connection refused errors after upgrade
 type: Troubleshooting
```
Comment (piotrmsc) [inline docs/service-mesh/10-03-connection-refused.md]: Nit, maybe change title? It was valid for kyma 1.0. Such errors may now be not only after upgrade but also after fresh installation

File: docs/service-mesh/10-02-no-access.md [hunk 12]
```diff
@@ -7,16 +7,37 @@ The `503` status code received when you try to access the Console UI or any othe
...
+     ```
+
+If this solution doesn't work, you need to change the image of the Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside them. Follow this steps:
+
+1. Edit the Istio Ingress Gateway Deployment:
+
+    ```bash
+    kubectl scale --replicas 0 -n istio-system deploy/istio-ingressgateway
+    kubectl edit deployment -n istio-system istio-ingressgateway
+    kubectl scale --replicas 1 -n istio-system deploy/istio-ingressgateway
```
Comment (piotrmsc) [inline docs/service-mesh/10-02-no-access.md]: Hmm, why do you need to scale to 0 and then to 1? Just do kubectl edit directly. Afterward, you will have only one replica

File: docs/service-mesh/10-03-connection-refused.md [hunk 13]
```diff
@@ -3,7 +3,9 @@ title: Connection refused errors after upgrade
...
 ---
 
-Starting with Kyma 1.0, mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. Alternatively, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
+Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. If you don't want to use a sidecar, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).
```
Comment (piotrmsc) [inline docs/service-mesh/10-03-connection-refused.md]: I would add here an example like `Connection reset by peer` etc

**Pull request merged**